src/PageForServer.ts

```diff
@@ -55,14 +55,16 @@ export class PageForServer extends Page {
     });
   }
 
-  renderStatic(status?: number, ns?: string) {
+  renderStatic(ns: string): void;
+  renderStatic(status: number, ns?: string): void;
+  renderStatic(status?: number | string, ns?: string) {
```

Comment:
@craigbeck to be honest I'm not sure if it makes sense for this 3rd overload to have an optional first arg. The only reason I marked it as optional is because the original signature showed both args as optional.

If we should always at least pass one arg, then I would update to something like:

```
  renderStatic(ns: string): void;
  renderStatic(status: number, ns?: string): void;
  renderStatic(arg1: number | string, ns?: string) {
```

Reply:
In derby@3, we changed `PageForServer#render`, normalize the order of params to be `render(ns: string, status?: number)`:
https://github.com/derbyjs/derby/releases/tag/v3.0.0

Longer term, we should aim to keep `renderStatic` consistent with that.

Unfortunately, we can only remove the `renderStatic(status: number, ns: string)` signature in a major version, so for now we'd need to support both signatures, with runtime type detection.

Perhaps something like this?
```ts
  renderStatic(ns: string, status?: number): void;
  renderStatic(status: number, ns: string): void;
  renderStatic(arg0: string | number, arg1?: number | string) {
    const ns = typeof arg0 === 'string' ? arg0 : arg1;
    const status = typeof arg1 === number ? arg1 : arg0;
    if (!ns) throw new Error('renderStatic ns param required');
```

Reply:
Thanks @ericyhwang - how would you recommend narrowing the type of ns to string? (I think it will still be typed as `string | number` after the existence check, and must be a string for the assignment of pageHtml and tailHtml below)

Maybe throw an additional error if ns is not a string? Or change the existence check to a type check?

Reply:
Also is it okay that in the new overloads there is no overload where both arguments are marked optional, as it is before these changes?